lib/AST/Pattern.cpp

```diff
@@ -68,6 +94,26 @@ StringRef Pattern::getKindName(PatternKind K) {
   llvm_unreachable("bad PatternKind");
 }
 
+StringRef Pattern::getDescriptivePatternKindName(DescriptivePatternKind K) {
+#define ENTRY(Kind, String) case DescriptivePatternKind::Kind: return String
+  switch (K) {
+    ENTRY(Paren, "paren");
+    ENTRY(Tuple, "tuple");
+    ENTRY(Named, "named");
+    ENTRY(Any, "any");
+    ENTRY(Typed, "typed");
+    ENTRY(Is, "is");
+    ENTRY(EnumElement, "enum-element");
+    ENTRY(OptionalSome, "optional-some");
+    ENTRY(Bool, "bool");
+    ENTRY(Expr, "expr");
```

Comment:
Could you preserve the existing printing behavior for these cases? What I mean by that is that they should use the same strings that are currently being produced from the `swift::operator<<(llvm::raw_ostream &OS, PatternKind kind)` method in this file. Once done, you can then delete that method.

Reply:
It would also be great if you could change `"optional .Some matching pattern"` to `"optional pattern"` while you're here :) The case hasn't been called `.Some` in a long while, and I think `"optional pattern"` is clear enough (that's its official name anyway).